src/test/java/com/ably/kafka/connect/DefaultAblyBatchClientTest.java

```diff
@@ -0,0 +1,68 @@
+package com.ably.kafka.connect;
+
+import com.ably.kafka.connect.client.DefaultAblyBatchClient;
+import com.ably.kafka.connect.config.ChannelConfig;
+import com.ably.kafka.connect.config.ChannelSinkConnectorConfig;
+import com.ably.kafka.connect.config.ConfigValueEvaluator;
+import com.ably.kafka.connect.config.DefaultChannelConfig;
+import com.ably.kafka.connect.mapping.ChannelSinkMapping;
+import com.ably.kafka.connect.mapping.DefaultChannelSinkMapping;
+import com.ably.kafka.connect.mapping.DefaultMessageSinkMapping;
+import com.ably.kafka.connect.mapping.MessageSinkMapping;
+import io.ably.lib.types.Message;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class DefaultAblyBatchClientTest {
+
+    @Test
+    public void testGroupMessagesByChannel() {
```

Comment:
This test is quite basic. 

We should really have a test that exercises the paths that hits the constraints such as max messages on a channel (based on the max KiB payload limit), and max messages in the total batch.  This will ensure we know that batches are constructed in a way that will always succeed.

Reply:
This will be addressed in the DLQ PR.